doc/releases/changelog-dev.md

```diff
@@ -6,6 +6,9 @@
 
 <h3>Improvements 🛠</h3>
 
+* The higher order primitives in program capture can now accept inputs with abstract shapes.
+  [(#6786)](https://github.com/PennyLaneAI/pennylane/pull/6786)
+
```

Comment:
Don't forget to add your name to the changelog 😄

Reply:
added.  thanks for the reminder.